### ISSUE TITLE
Unable to compile sdbg version using gfortran 8.3.0

### ISSUE THREAD

**Issue by michaellass** [cp2k/cp2k#887]
I am currently struggling to build the sdbg version of CP2K using gfortran 8.3.0. I get the following errors:

```
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0:

       TYPE(dbcsr_t_type), DIMENSION(1, 1)                :: t_3c_int_batched
 
Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 72B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 80B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 152B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 160B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 216B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 224B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 280B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 288B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 344B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 352B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 408B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 416B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 496B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 504B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 576B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 584B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 640B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 648B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 704B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 712B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 768B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 776B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 832B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 840B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 904B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 912B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 968B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 976B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1032B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1040B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1096B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1104B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1160B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1168B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1224B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1232B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1304B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1312B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1384B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1392B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1448B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1456B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1512B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1520B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1576B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1584B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1640B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1648B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1736B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1744B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1800B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1808B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1864B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1872B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1928B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:355:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.479 + 1936B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0:

       TYPE(dbcsr_t_type), DIMENSION(1, 1)                :: t_3c_int_mo_1, t_3c_int_mo_2
 
Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 72B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 80B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 152B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 160B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 216B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 224B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 280B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 288B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 344B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 352B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 408B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 416B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 496B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 504B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 576B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 584B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 640B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 648B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 704B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 712B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 768B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 776B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 832B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 840B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 904B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 912B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 968B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 976B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1032B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1040B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1096B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1104B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1160B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1168B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1224B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1232B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1304B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1312B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1384B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1392B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1448B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1456B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1512B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1520B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1576B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1584B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1640B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1648B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1736B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1744B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1800B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1808B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1864B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1872B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1928B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.264 + 1936B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0:

       TYPE(dbcsr_t_type), DIMENSION(1, 1)                :: t_3c_int_mo_1, t_3c_int_mo_2
 
Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 72B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 80B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 152B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 160B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 216B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 224B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 280B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 288B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 344B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 352B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 408B]’ is used uninitialized in this function [-Werror=uninitialized]
/scratch/hpc-prf-ldft/lass/cp2k.plain/src/hfx_ri.F:977:0: Error: ‘MEM[(struct dbcsr_t_type *)&dbcsr_t_type.260 + 416B]’ is used uninitialized in this function [-Werror=uninitialized]
f951: some warnings being treated as errors
make[3]: *** [hfx_ri.o] Error 1
make[2]: *** [all] Error 2
make[1]: *** [sdbg] Error 2
make: *** [all] Error 2
```

I saw that there was quite some activity in dbcsr's develop branch wrt. tensors, so I tried building with https://github.com/cp2k/dbcsr/commit/0046876a3d42d9505c7c52f7c2c0512ced4b5929 but I get the same result.

Interestingly, the dashboard currently does not show any build failures. Any idea what the problem could be here?

**Comment by pseewald** [comment]
This error looks like a compiler bug to me. Either use a different version of gfortran or maybe compilation without `-Werror=uninitialized` works.

**Comment by michaellass** [comment]
Yes, without `-Werror=uninitialized` compilation is successful. I guess it will work with gfortran 9.2 because this is what the sdbg CI runners use.

**Comment by dev-zero** [comment]
unfortunately this is something we've also seen on our systems on gcc 8.3+ in different places of the code

**Comment by dev-zero** [comment]
Please stop closing valid/unresolved issues. Several people in our group have to modify the generated `.sdbg` arch file to make it build.

**Comment by oschuett** [comment]
I'm sorry. It appeared to me the matter was settled.

Our debug arch files only work with the current toolchain setup, because they are optimized for catching bugs.

So, I'd suggest we add a note to [INSTALL.md](https://github.com/cp2k/cp2k/blob/master/INSTALL.md) and close this issue again.

**Comment by dev-zero** [comment]
I reduce the test case for the issue we see in CP2K with gcc from our toolchain and submitted it first here as a comment:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=77504

But it got its own issue now:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=96312

It seems it is not just a false positive but rather an actual bug in the gcc frontend.

**Comment by mkrack** [comment]
Resolved via #1017
